apply-roles

```diff
@@ -9,125 +9,141 @@ readonly BRANCH="master"
 readonly GITHUB_URI="https://raw.githubusercontent.com/cloudconformity/azure-onboarding-scripts/${BRANCH}"
 readonly CUSTOM_ROLE_NAME="Custom Role - Cloud One Conformity"
 
-application_id=
-subscription_id=
+apply_roles_to_subscription() {
+  local subscription_id="${1}"
 
-# capture CLI arguments
-while [ "${1}" != "" ]; do
-  case "${1}" in
-  -a | --application-id)
-    shift
-    application_id="${1}"
-    ;;
-  -s | --subscription-id)
-    shift
-    subscription_id="${1}"
-    ;;
-  esac
-  shift
-done
+  echo "Checking custom role assignment for subscription - ${subscription_id}"
+  is_custom_role_assigned=$(az role assignment list \
+    --role "${reader_role_id}" \
```

Comment:
❓ Question for reviewers...
These variables are using the Global scoped variables. Would it be better to pass all of these in as arguments to the function?

Reply:
@iaindurham Given the relatively annoying way of accessing function arguments in shell scripts, I'd say either one is fine.